src/Drips.sol

```diff
@@ -666,22 +698,23 @@ library Drips {
     /// @param amtPerSec The dripping rate
     function _addDelta(
         mapping(uint32 => AmtDelta) storage amtDeltas,
-        uint32 cycleSecs,
-        uint32 timestamp,
+        uint256 cycleSecs,
+        uint256 timestamp,
         int256 amtPerSec
     ) private {
         unchecked {
-            AmtDelta storage amtDelta = amtDeltas[_cycleOf(timestamp, cycleSecs)];
+            AmtDelta storage amtDelta = amtDeltas[_cycleOf(uint32(timestamp), uint32(cycleSecs))];
             int256 thisCycleDelta = amtDelta.thisCycle;
             int256 nextCycleDelta = amtDelta.nextCycle;
 
             // In order to set a delta on a specific timestamp it must be introduced in two cycles.
             // The cycle delta is split proportionally based on how much this cycle is affected.
             // The next cycle has the rest of the delta applied, so the update is fully completed.
-            uint32 nextCycleSecs = timestamp % cycleSecs;
-            uint32 thisCycleSecs = cycleSecs - nextCycleSecs;
-            thisCycleDelta += int256(uint256(thisCycleSecs)) * amtPerSec;
-            nextCycleDelta += int256(uint256(nextCycleSecs)) * amtPerSec;
+            int256 amtPerCycle = (amtPerSec * int256(cycleSecs)) / int256(AMT_PER_SEC_MULTIPLIER);
+            int256 amtSplitCycle = (amtPerSec * int256(timestamp % cycleSecs)) /
```

Comment:
The helper would be:
```solidity
function roundDown(int256 amt) returns (int256) {
    return amt / int256(AMT_PER_SEC_MULTIPLIER);
}
```
I think that a helper like this would increase noise instead of improving readability. It encapsulates very little logic and has a lot of usage-specific type juggling. Finally, it can be used in only 2 places in the entire codebase.

Reply:
I think this is personal taste. For me it would be much more clear that a rounding is happening here. 
The reader might assume it is converting it back to the base system without realizing a cut-off happens here. 

No strong opinion, however.